crates/rust-analyzer/src/reload.rs

```diff
@@ -769,7 +772,11 @@ impl GlobalState {
     pub(super) fn fetch_build_data_error(&self) -> Result<(), String> {
         let mut buf = String::new();
 
-        for ws in &self.fetch_build_data_queue.last_op_result().1 {
+        let Some((_, ws)) = &self.fetch_build_data_queue.last_op_result() else {
```

Comment:
Maybe consider `.unwrap_or_default()` here? That way the code could execute the `for` loop as below, and it's maybe easier to reason about (fewer returns).

Reply:
the unfortunate bit is that `Option` is also an iterator, so that'd lead to some very weird combinator gunk for a nested for loop. I'd prefer to keep this as-is.

Reply:
I prefer the early return here as it makes the logic more clear imo. an "early return control flow prelude" in a function is not that bad for reasoning if you ask me (unlike returns in the middle of a function).